tests/foreman/api/test_ansible.py

```diff
@@ -602,3 +602,102 @@ def test_negative_ansible_job_timeout_to_kill(
+    @pytest.mark.no_containers
+    @pytest.mark.rhel_ver_match('8')
+    @pytest.mark.parametrize('setting_update', ['remote_execution_effective_user'], indirect=True)
+    def test_positive_privilege_escalation_ansible_playbook(
+        self,
+        target_sat,
+        rhel_contenthost,
+        module_org,
+        module_location,
+        module_activation_key,
+        setting_update,
+    ):
+        """Privilege escalation defined inside ansible playbook tasks is working
+        when executing the playbook via Remote Execution
+
+        :id: 8c63fd1a-2121-4cce-9ec1-ae12817c9cc4
+
+        :steps:
+            1. Register a RHEL host to Satellite.
+            2. Setup a user on that host.
+            3. Create a playbook.
+            4. Set the SSH user to the created user, and unset the Effective user.
+            5. Run the playbook.
+
+        :expectedresults: In the playbook, created user is expected instead root user.
+        """
+        playbook = '''
+                    ---
+                    - name: Test Play
+                      hosts: all
+                      gather_facts: false
+                      tasks:
+                        - name: Check current user
+                          command: bash -c "whoami"
+                          register: def_user
+                        - debug:
+                            var: def_user.stdout
+                        - name: Check become user
+                          command: bash -c "whoami"
+                          become: true
+                          become_user: testing
+                          register: bec_user
+                        - debug:
+                            var: bec_user.stdout
+                '''
+        result = rhel_contenthost.register(
+            module_org, module_location, module_activation_key.name, target_sat
+        )
+        assert result.status == 0, f'Failed to register host: {result.stderr}'
+        assert rhel_contenthost.username == 'root'
+        user1 = rhel_contenthost.execute('useradd rexuser')
+        assert user1.status == 0
+        echo1 = rhel_contenthost.execute('echo dog8code | passwd rexuser --stdin')
+        assert 'passwd: all authentication tokens updated successfully.' in echo1.stdout
+        pswd = rhel_contenthost.execute(
+            'echo "rexuser   ALL=NOPASSWD:   ALL" | tee -a /etc/sudoers.d/rexuser'
+        )
+        assert pswd.status == 0
+        user2 = rhel_contenthost.execute('useradd testing')
+        assert user2.status == 0
+        echo2 = rhel_contenthost.execute('echo dog8code | passwd testing --stdin')
+        assert 'passwd: all authentication tokens updated successfully.' in echo2.stdout
+        whoami = rhel_contenthost.execute('su - testing -c "whoami"')
+        assert 'testing' in whoami.stdout or whoami.status == 0
+        setting_update.value = ''
+        setting_update = setting_update.update({'value'})
+        assert setting_update.value == ''
```

Comment:
Instead of updating remote_execution_effective_user setting at satellite, why not using effective_user from the REX job create page?

Reply:
There is no option available in Nailgun to update the SSH user remotely. So using `setting_update` updating the value.

Reply:
yes, effective_user is not available but you can add it anyway, its already part of apidoc

Reply:
I tried it before, but it didn't give the proper output for two users.